### PR TITLE
Swithc to using build.sbt & update dependency versions.

### PR DESCRIPTION
**Pull request by jatcwang**
Just experimenting with http4s and manage to get the demo working at newest version of various dependencies. Only thing I didn't try is publishing using native-packager.

### CODE REVIEW

**Review by bryce-anderson — COMMENTED** [review]
Thanks for taking a look at this! Is been a while since it's been updated, hadn't it? 

From my phone or looks good, but I should get to a command line before merging.

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name := "http4s-demo"
+
+version := "1.0-SNAPSHOT"
+
+scalaVersion := "2.11.7"
+
+val rhoVersion = "0.12.0"
+
+val http4sVersion = "0.14.7"
```
Comment (rossabaker) [inline build.sbt]: I just released 0.14.8 around the same time you were submitting this.

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name := "http4s-demo"
+
+version := "1.0-SNAPSHOT"
+
+scalaVersion := "2.11.7"
+
+val rhoVersion = "0.12.0"
```
Comment (rossabaker) [inline build.sbt]: Should we be using the "a" versions of this and http4s to be on scalaz-7.2?  I guess our naming scheme implies the scalaz-7.1 version as a "default," but that's unfortunate.

**Review by rossabaker — COMMENTED** [review]
Thanks, @jtcwang.  

It loads for me.  Everything works except the Swagger UI.  This could be a quirk on localhost.  The swagger.json itself renders.

Looks good to me, unless you want to clean up the version comments above.

**Comment by jtcwang** [comment]
Updated to http4s 0.14.8a  & rho 0.12.0a for scalaz 7.2

Also gave updated sbt-native-package and tried `packageBin`. Works fine for me  :smile:

**Comment by bryce-anderson** [comment]
Compiles for me, too! I do think there is one thing missing: a 
```scala
enablePlugins(JavaAppPackaging)
```
at the top of the build.sbt file.

**Comment by jtcwang** [comment]
Done!

**Comment by bryce-anderson** [comment]
Looks great! Thanks @jtcwang.

**Pull request merged**